### ISSUE TITLE
[UPRISE]After rereading the paper UPRISE: Universal Prompt Retrieval for Improving Zero-Shot Evaluation,I have some questions.

### ISSUE THREAD

**Issue by zhouchang123** [microsoft/LMOps#262]
![image](https://github.com/user-attachments/assets/f413c84a-cc4d-44e7-80fb-1d9c160a7c06)
1.How to get the scores through GPT-Neo-2.7B?
2.In which procedure,the prompt get positive or negative,after get the scores  or after encode before score?

**Comment by cdxeve** [comment]
Q1: How to get the scores through GPT-Neo-2.7B?
By calculating the task metric score of each input concatenation of prompt + testing input, see Section 3.2.

Q2: In which procedure, the prompt get positive or negative, after get the scores or after encode before score?
After getting the scores. `For all the scored prompts for a training example, we label the prompt with the highest score as positive. For negative samples, we randomly sample B training demonstrations from the prompt pool, in addition, we label B demonstrations corresponding to the lowest B scores in the sampled prompts as hard negatives`, details are in Section 3.2.

**Comment by zhouchang123** [comment]
What about the score through prompt retriever?Is it same as GPT-Neo-2.7B?
Thanks very much.

**Comment by cdxeve** [comment]
You may refer to Section 3.4 to see how we get the score after tuning the prompt retriever.

**Comment by zhouchang123** [comment]
Section 3.4 introduced the inference part?
It is the same in training pipline ?
![image](https://github.com/user-attachments/assets/6b964276-7d1e-4418-af82-33e4561c72ee)

**Comment by cdxeve** [comment]
Training is in Section 3.3, you may refer to the provided code as well.

**Comment by zhouchang123** [comment]
![image](https://github.com/user-attachments/assets/5a10b223-e86e-4c98-bfbd-d2bdad604cd8)
Section 3.3 only introduce sim(x, p) ,do you mean sim(x, p) is the score ?
![image](https://github.com/user-attachments/assets/c10ce61c-0606-42c8-b24d-76dfd265a6e0)

**Comment by cdxeve** [comment]
Yes, sim(x, p) is the score.

**Comment by zhouchang123** [comment]
In paper,the positive prompt is number 1 and negative prompt number is  20.But not demonstrate the total number of prompts.
What will happen if the prompts not positive or negative?
To the prompts not positive or negative,InfoNCE seems not include these prompts.

**Comment by cdxeve** [comment]
Yes, InfoNCE would not consider the prompts that are neither positive nor negative.

**Comment by zhouchang123** [comment]
I found some confusion about the pipline of training and inferencing.
In training pipline, the input is include the task name and the query and the metric considerates the task.
However when inferencing,the input is only the query without task name.
So could add a module that according to the query to clarify its task name,and first filter the task name then retriever? @cdxeve

**Comment by cdxeve** [comment]
We do not input the task name during training, and the task name in the image is only for ease of understanding. You may refer to the formula in section 3.2 for details.

**Comment by zhouchang123** [comment]
I viewed the file prompt_pool.json and each dict is annotated to different task name.So the task name is only to divide to its metric score?
The normal state of mind when retrieving is to retriever in the prompts of similar task rather than all the prompts.
